packages/dds/ordered-collection/src/consensusOrderedCollection.ts

```diff
@@ -100,6 +101,10 @@ export class ConsensusOrderedCollection<T = any>
 	 * The set of values that have been acquired but not yet completed or released
 	 */
 	private jobTracking: JobTrackingInfo<T> = new Map();
+	private get orderedCollectionLogger() {
+		// eslint-disable-next-line @typescript-eslint/no-unsafe-return
+		return createChildLogger({ logger: this.runtime.logger });
```

Comment:
2 pieces of feedback for this:
1. We should probably save the child logger as a private property on construction. Otherwise we will generate a new logger every time this getter is called.
2. It would be good to add a type-narrowing helper function to tell if the logger we have is already a `ITelemetryLoggerExt`. If it is, we don't need to call `createChildLogger`. We could add an `isTelemetryLoggerExt` function that takes in a `ITelemetryBaseLogger` and returns whether or not it is a `ITelemetryLoggerExt`. See https://github.com/microsoft/FluidFramework/blob/main/packages/utils/telemetry-utils/src/config.ts#L333 for an example of how to do this.

Reply:
`createChildLogger()` already does a check kind of like that, but I'm not sure "if it is, we don't need to call `createChildLogger()`" is correct. Calling the function is guaranteed to return a different logger object than the base logger that gets passed in, which means that changes to the base logger are not reflected in the new child one. We could optimize to simply return the base logger that is passed if there is nothing to add to it (no additional namespace, no additional properties) but then if that base logger gets changed then the child logger would see those changes too, when today it does not. @anthony-murphy when you wrote `createChildLogger()`, was it by design that we always return a new object (even if it's a "snapshot copy" of the base logger)? Or should we consider the additional optimization of immediately returning the base logger if it's already a ChildLogger and no additional namespace or props were specified? cc @markfields for thoughts too.

Reply:
Seems like it's safe to "snapshot" `this.runtime.logger` with a `createChildLogger()` call. It's really [an accessor for the datastoreRuntime's mc.logger](https://github.com/microsoft/FluidFramework/blob/218b985f5708c781ee31faaf639209c2a0de0fc6/packages/runtime/datastore/src/dataStoreRuntime.ts#L183-L185), and `this.mc` is only assigned [here](https://github.com/microsoft/FluidFramework/blob/218b985f5708c781ee31faaf639209c2a0de0fc6/packages/runtime/datastore/src/dataStoreRuntime.ts#L233-L239), which already results in a `createChildLogger()` call. I'll go ahead with that change.